src/vm/node_modules/VM.js

```diff
@@ -518,6 +518,38 @@ function getZpools(log, callback)
     });
 }
 
+function getOptimalVolblocksize(log, callback)
+{
+    var args = ['-C'];
+    var cmd = '/usr/sbin/zdb';
+
+    assert(log, 'no logger passed to zpoolRequiresLargerVolblocksize()');
+
+    traceExecFile(cmd, args, log, 'zdb-c',
```

Comment:
How will this work on a multi-pool installation?  Imagine someone with a standalone SmartOS with a dedicated bootable pool and `zones`?  Should you narrow it to $SYSPOOL ?

Reply:
`zdb -C $SYSPOOL` ==> I'm not sure where VM.js gets what the illumos-joyent scripts call `$SYSPOOL` but that's what you want.  Might even just be able to pull the first component in the ZFS dataset name.

Reply:
Actually you want to do the pool the image/instance resides on, since images can stored on a pool declared at import time.

To make matters more complex, standalone SmartOS allows kvm to have disks be on separate pools. disk0 can be on `$SYSPOOL`, where the image is and disk1 can be declared to be on an alternate pool. But I think at this point in the code, you're just checking for *a single* disk/volume, so you can get the zpool property of the disk object.

Reply:
Please check if the new approach is acceptable.